File: bun/src/lib/priorities.ts
export const PRIORITY_MAP: Record<string, number> = {
  none: 0,
  urgent: 1,
  high: 2,
  medium: 3,
  low: 4,
};

export const PRIORITY_VALUES = "none | urgent | high | medium | low";

/**
 * Resolve a priority string to its numeric value.
 * Returns undefined if the input is not a valid priority name.
 */
export function resolvePriority(input: string): number | undefined {
  return PRIORITY_MAP[input.toLowerCase()];
}

File: internal/priorities/priorities.go
package priorities

import "strings"

var Map = map[string]int{
	"none":   0,
	"urgent": 1,
	"high":   2,
	"medium": 3,
	"low":    4,
}

const Values = "none | urgent | high | medium | low"

// Resolve returns the priority number for a name, or -1 if unrecognized.
func Resolve(input string) (int, bool) {
	p, ok := Map[strings.ToLower(input)]
	return p, ok
}
